core/src/main/java/org/testcontainers/containers/VncRecordingContainer.java

```diff
@@ -87,19 +88,27 @@ protected void configure() {
         );
     }
 
+    @SneakyThrows
+    public void saveRecordingToFile(File file) {
+        try(InputStream inputStream = streamRecording()) {
+            Files.copy(inputStream, file.toPath(), StandardCopyOption.REPLACE_EXISTING);
+        }
+    }
+
     @SneakyThrows
     public InputStream streamRecording() {
+        reencodeRecording();
+
         TarArchiveInputStream archiveInputStream = new TarArchiveInputStream(
                 dockerClient.copyArchiveFromContainerCmd(getContainerId(), RECORDING_FILE_NAME).exec()
         );
         archiveInputStream.getNextEntry();
         return archiveInputStream;
     }
 
-    @SneakyThrows
-    public void saveRecordingToFile(File file) {
-        try(InputStream inputStream = streamRecording()) {
-            Files.copy(inputStream, file.toPath(), StandardCopyOption.REPLACE_EXISTING);
-        }
+    private void reencodeRecording() throws IOException, InterruptedException {
+        String newFlvOutput = "/newScreen.flv";
+        execInContainer("ffmpeg" , "-i", RECORDING_FILE_NAME, "-vcodec", "libx264", newFlvOutput);
```

Comment:
Since commenting doesn't seem to reopen the issue, I have to make a new one https://github.com/testcontainers/testcontainers-java/pull/3180/files#r524394794

```suggestion
        String newFlvOutput = "/newScreen.mp4";
        execInContainer("ffmpeg" , "-i", RECORDING_FILE_NAME, "-vcodec", "libx264", "-movflags", "faststart", newFlvOutput);
```